### PR TITLE
6.0 initial

### PR DESCRIPTION
**Pull request by tmds**
This PR targets branch `dev_60` that contains a copy of our `5.0` directory [with `5` replaced with `6`](https://github.com/redhat-developer/s2i-dotnetcore/commit/4e414fd3a15bafc5b2961e8f85f3473c9603dc8d).

The PR removes the CentOS 7 and RHEL 7 images and updates the UBI 8 and Fedora images to use Microsoft's preview7 build.

All tests pass except for `DOTNET_PUBLISH_READYTORUN`, which I think fails due to https://github.com/dotnet/sdk/issues/18999.

### CODE REVIEW

File: 6.0/build/Dockerfile.fedora [hunk 1]
```diff
@@ -25,12 +25,14 @@ USER 0
 # Copy the S2I scripts from the specific language image to $STI_SCRIPTS_PATH.
 COPY ./s2i/bin/ /usr/libexec/s2i
 
-RUN INSTALL_PKGS="npm dotnet-sdk-6.0 rsync procps-ng findutils" && \
-    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
-    rpm -V $INSTALL_PKGS && \
-    yum clean all -y && \
-# yum cache files may still exist (and quite large in size)
-    rm -rf /var/cache/yum/*
+# TODO: use RHEL packages.
```
Comment (omajid) [inline 6.0/build/Dockerfile.fedora:28]: RHEL -> Fedora ?

File: 6.0/runtime/Dockerfile.fedora [hunk 2]
```diff
@@ -46,12 +46,19 @@ COPY ./root/usr/bin /usr/bin
...
+#     rpm -V $INSTALL_PKGS && \
+#     yum clean all -y && \
+# # yum cache files may still exist (and quite large in size)
+#     rm -rf /var/cache/yum/*
+RUN curl https://download.visualstudio.microsoft.com/download/pr/ce3fd989-b69d-439a-9cac-09ad40597db8/2848d49480b6e7b1b2a18cfa46d724e2/dotnet-sdk-6.0.100-preview.7.21379.14-linux-x64.tar.gz -o /tmp/dotnet.tar.gz && \
+    mkdir /opt/dotnet && \
+    tar -xf /tmp/dotnet.tar.gz -C /opt/dotnet && \
+    ln -s /opt/dotnet/dotnet /usr/bin/dotnet && \
+    yum install -y nss_wrapper tar unzip findutils libicu
+ENV DOTNET_ROOT=/opt/dotnet
```
Comment (omajid) [inline 6.0/runtime/Dockerfile.fedora:61]: I am surprised we don't need to modify `PATH` too.

Comment (tmds) [reply 6.0/runtime/Dockerfile.fedora:61]: `ln -s /opt/dotnet/dotnet /usr/bin/dotnet` is what you are looking for.

**Comment by tmds** [comment]
All tests pass now. I'm going to merge this to the `dev_60` branch. Other changes (like updating to our own packages) can be made by targeting that branch.

**Pull request merged**